src/Utils/Models.php

```diff
@@ -0,0 +1,291 @@
+<?php
+
+declare(strict_types=1);
+
+namespace WordPress\AiClient\Utils;
+
+use WordPress\AiClient\Providers\Models\Contracts\ModelInterface;
+use WordPress\AiClient\Providers\Models\DTO\ModelRequirements;
+use WordPress\AiClient\Providers\Models\Enums\CapabilityEnum;
+use WordPress\AiClient\Providers\Models\ImageGeneration\Contracts\ImageGenerationModelInterface;
+use WordPress\AiClient\Providers\Models\SpeechGeneration\Contracts\SpeechGenerationModelInterface;
+use WordPress\AiClient\Providers\Models\SpeechGeneration\Contracts\SpeechGenerationOperationModelInterface;
+use WordPress\AiClient\Providers\Models\TextGeneration\Contracts\TextGenerationModelInterface;
+use WordPress\AiClient\Providers\Models\TextToSpeechConversion\Contracts\TextToSpeechConversionModelInterface;
+use WordPress\AiClient\Providers\Models\TextToSpeechConversion\Contracts\TextToSpeechConversionOperationModelInterface;
+use WordPress\AiClient\Providers\ProviderRegistry;
+
+/**
+ * Utility class for model discovery and interface validation.
+ *
+ * Combines model auto-discovery capabilities with interface validation
+ * to provide a unified model utility service.
+ *
+ * @since n.e.x.t
+ */
+class Models
```

Comment:
What are the methods in this class for? 

I don't see them used anywhere, and they don't make sense to me. We need to find models based on _all_ `ModelRequirements`, i.e. based on the `list<Message>` and `ModelConfig` instances, not just on a single capability like these methods do it.

Maybe I'm missing something, but to me these methods seem neither right nor necessary (at this point at least). 🤔

Reply:
This should be unnecessary in light of the `PromptBuilder` changes.